tests/0148-offset_fetch_commit_error_mock.c

```diff
@@ -0,0 +1,548 @@
+
+        SUB_TEST_PASS();
+}
+
+static void log_cb_closing_issue(const rd_kafka_t *rk,
+                                 int level,
+                                 const char *fac,
+                                 const char *buf) {
+        if (strstr(buf, "Acknowledging target assignment")) {
+                mtx_lock(&log_lock);
+                ack_target_assignment_count++;
+                if (ack_target_assignment_count == 2) {
+                        trigger_consumer_close = rd_true;
+                        cnd_signal(&log_cnd);
+                }
+                mtx_unlock(&log_lock);
+        }
+        if (strstr(buf, "unable to OffsetCommit") &&
+            strstr(buf, "Broker: The member epoch is stale")) {
+                mtx_lock(&log_lock);
+                got_stale_member_epoch_error++;
+                mtx_unlock(&log_lock);
+        }
+}
+
+/**
+ * @brief This test checks that when a consumer acknowledges revocations and
+ * sends a heartbeat just before leaving the group, the heartbeat response may
+ * still be in flight while the leave process begins. In this scenario, the
+ * heartbeat response must update the member epoch rather than being discarded.
+ * Otherwise, subsequent commit requests (which will be required for leaving)
+ * may fail with a stale member epoch error.
+ */
+void do_test_consumer_group_heartbeat_ack_assignment_close(
+    rd_kafka_mock_cluster_t *mcluster,
+    const char *bootstraps) {
+
+        char topic1[256], topic2[256];
+        rd_kafka_t *producer, *consumer;
+        rd_kafka_conf_t *conf, *producer_conf;
+        uint64_t testid  = test_id_generate();
+        const int msgcnt = 5;
+        test_msgver_t mv;
+        const char *debug_contexts[3] = {"cgrp", NULL};
+        int64_t close_start, close_end;
+        const int session_timeout_ms = 3000;
+        const int heartbeat_rtt_ms   = 200;
+
+        SUB_TEST_QUICK();
+
+        mtx_init(&log_lock, mtx_plain);
+        cnd_init(&log_cnd);
+
+        strcpy(topic1, test_mk_topic_name("topic1", 1));
+        strcpy(topic2, test_mk_topic_name("topic2", 1));
+
+        test_conf_init(&conf, NULL, 30);
+        test_conf_set(conf, "bootstrap.servers", bootstraps);
+        test_conf_set(conf, "group.protocol", "consumer");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "enable.auto.commit", "true");
+        test_conf_set(conf, "auto.commit.interval.ms", "2000");
+        test_conf_set(conf, "fetch.wait.max.ms", "100");
+        test_conf_set_log_interceptor(conf, log_cb_closing_issue,
+                                      debug_contexts);
+
+        rd_kafka_mock_coordinator_set(mcluster, "group", topic1, 1);
+        rd_kafka_mock_set_group_consumer_session_timeout_ms(mcluster,
+                                                            session_timeout_ms);
+        rd_kafka_mock_set_group_consumer_heartbeat_interval_ms(mcluster, 500);
+
+        // Producer Initialization
```

Comment:
Let's use C style comments instead of C++ ones even if the compiler accepts it
```suggestion
        /* Producer Initialization */
```